pinot-spi/src/main/java/org/apache/pinot/spi/stream/v2/Checkpoint.java

```diff
@@ -0,0 +1,6 @@
+package org.apache.pinot.spi.stream.v2;
+
+public interface Checkpoint {
```

Comment:
We agreed to rename this. 
Also, we don't need stream.v2. We can add this in the existing stream interfaces.